gdk/wizard/Prompter.py

```diff
@@ -0,0 +1,259 @@
+        self.checker = WizardChecker()
+        self.parser = argparse.ArgumentParser()
+
+    def prompter(self, field, required, max_attempts=3):
+        """
+        Prompts the user for a value of a given field key
+
+        Parameters
+        ----------
+            field (enum): a field key of the gdk_config file to be changed
+            value (string): the current value corresponding the field key to be changed
+            required (boolean): if the field is a required field of the gdk config file
+            max_attempts (int): the maximum number of attempts to get a valid response from the user
+
+        Returns
+        -------
+            string: the value for the field key "field"
+        """
+        current_field_value = self.data.get_field(field)
+        require = "REQUIRED " if required else "OPTIONAL "
+        link = "https://docs.aws.amazon.com/greengrass/v2/developerguide/gdk-cli-configuration-file.html#gdk-config-format"
+        for attempt in range(1, max_attempts + 1):
+            parser_argument = field.value.key
+            if field == ConfigEnum.BUILD_OPTIONS:
+                parser_argument = "build_options"
+            elif field == ConfigEnum.PUBLISH_OPTIONS:
+                parser_argument = "publish_options"
+
+            args = self.parser.parse_args(
+                [
+                    f"--{parser_argument}",
+                    self.interactive_prompt(
+                        parser_argument, current_field_value, require
+                    ),
+                ]
+            )
+            response = getattr(args, parser_argument).strip()
+
+            # only way customer is asked about custom_build_command if they have custom build system
+            # in which case they must specify a custom build command that is not None
+            if field == ConfigEnum.CUSTOM_BUILD_COMMAND:
+                if response == "None":
+                    print(
+                        f"Attempt {attempt}/{max_attempts}: Must Specify a custum build command.\nPlease vist: {link}"
+                    )
+                    continue
+
+            # if customer input response is the default value or the customer input response is
+            # the same as the current value, return the current value, then return the same value
+            if (
+                response == field.value.default
+                or response == current_field_value
+                or self.checker.is_valid_input(response, field)
+            ):
+                return response
+            print(
+                f"Attempt {attempt}/{max_attempts}: Invalid response. Please try again.\nPlease vist: {link}"
+            )
+
+        if field == ConfigEnum.CUSTOM_BUILD_COMMAND:
+            self.utils.write_to_config_file(self.field_dict, self.project_config_file)
+            sys.exit(
+                "You have failed to enter a valid custom build command. Exiting wizard..."
+            )
```

Comment:
Why do we need this?

Reply:
gotcha so if the program is still executing outside of the for loop it means that the response from the user is still not valid after the max number of tries. Thus, if its the case that wizard is prompting for custom build command then, we must exit the wizard instead of reprompting, if we're not prompting the wizard then we just take the current valid already in the field_map